src/neo/SmartContract/Native/Tokens/Nep5Token.cs

```diff
@@ -112,7 +115,7 @@ internal protected virtual void Burn(ApplicationEngine engine, UInt160 account,
             BigInteger totalSupply = new BigInteger(storage.Value);
             totalSupply -= amount;
             storage.Value = totalSupply.ToByteArrayStandard();
-            engine.SendNotification(Hash, new Array(new StackItem[] { "Transfer", account.ToArray(), StackItem.Null, amount }));
+            engine.SendNotification(tx, Hash, new Array(new StackItem[] { "Transfer", account.ToArray(), StackItem.Null, amount }));
```

Comment:
Does we need the tx inside the notification?

Reply:
![image](https://user-images.githubusercontent.com/12050350/72658925-b208a000-39f2-11ea-8167-9404c33ebd0e.png)
After deploying sc, the ScriptContainer is null in RpcNep5Tracker. we need to get the tx info to trace the balance change.

Reply:
I prefer to add it in `Blockchain.ApplicationExecuted` in order to prevent errors like https://github.com/neo-project/neo/issues/1410

Reply:
Do you mean we handle the gas burn for each transaction in `Nep5Tracker`, like this:
```
foreach (Blockchain.ApplicationExecuted appExecuted in applicationExecutedList)
{
    // Executions that fault won't modify storage, so we can skip them.
    if (appExecuted.VMState.HasFlag(VMState.FAULT)) continue;
    foreach (var notifyEventArgs in appExecuted.Notifications)
    {
        if (!(notifyEventArgs?.State is VM.Types.Array stateItems) || stateItems.Count == 0
            || !(notifyEventArgs.ScriptContainer is Transaction transaction))
            continue;
        HandleNotification(snapshot, appExecuted.Transaction, notifyEventArgs.ScriptHash, stateItems,
            nep5BalancesChanged, ref transferIndex);
    }

    // Handle SystemFee and NetworkFee GAS burn for each transaction
    if (!( appExecuted.Transaction is Transaction tx)) continue;
    HandleNotification(snapshot, appExecuted.Transaction, NativeContract.GAS.Hash,
        new VM.Types.Array(new VM.Types.StackItem[] { "Transfer", appExecuted.Transaction.Sender.ToArray(), VM.Types.StackItem.Null, tx.SystemFee + tx.NetworkFee }),
        nep5BalancesChanged, ref transferIndex);
}

```

Reply:
I see, let me thinking about

Reply:
```cs
// Handle SystemFee and NetworkFee GAS burn for each transactio
```

aheuaheauea, the prototype for burning all fees..

I do not if it is good to pass `tx` to the NotificationHandler, as Shargon mentioned. But I need to think as well.